nilearn/decomposition/dict_learning.py

```diff
@@ -0,0 +1,275 @@
+"""
+Dictionary learning estimator: Perform a map learning algorithm by learning
+a temporal dense dictionary along with sparse spatial loadings, that
+constitutes output maps
+"""
+
+# Author: Arthur Mensch
+# License: BSD 3 clause
+
+from __future__ import division
+
+import warnings
+from distutils.version import LooseVersion
+
+import numpy as np
+import sklearn
+from sklearn.base import TransformerMixin
+from sklearn.decomposition import dict_learning_online
+from sklearn.externals.joblib import Memory
+from sklearn.linear_model import Ridge
+
+from .base import BaseDecomposition, mask_and_reduce
+from .canica import CanICA
+from .._utils.cache_mixin import CacheMixin
+
+if LooseVersion(sklearn.__version__) >= LooseVersion('0.17'):
+    # check_input=False is an optimization available only in sklearn >=0.17
+    sparse_encode_args = {'check_input': False}
+
+
+def _compute_loadings(components, data):
+    ridge = Ridge(fit_intercept=None, alpha=1e-8)
+    ridge.fit(components.T, np.asarray(data.T))
+    loadings = ridge.coef_.T
+
+    S = np.sqrt(np.sum(loadings ** 2, axis=0))
+    S[S == 0] = 1
+    loadings /= S[np.newaxis, :]
+    return loadings
+
+
+class DictLearning(BaseDecomposition, TransformerMixin, CacheMixin):
+    """Perform a map learning algorithm based on component sparsity,
+     over a CanICA initialization.  This yields more stable maps than CanICA.
+
+    Parameters
+    ----------
+    mask: Niimg-like object or MultiNiftiMasker instance, optional
+        Mask to be used on data. If an instance of masker is passed,
+        then its mask will be used. If no mask is given,
+        it will be computed automatically by a MultiNiftiMasker with default
+        parameters.
+
+    n_components: int
+        Number of components to extract
+
+    n_epochs: float
+        Number of epochs the algorithm should run on the data
+
+    alpha: float, optional, default=1
+        Sparsity controlling parameter
+
+    dict_init: Niimg-like object, optional
+        Initial estimation of dictionary maps. Would be computed from CanICA if
```

Comment:
"initial estimation"?...perhaps more "Custom initilization of map parameters..."